python/ray/dag/input_node.py

```diff
@@ -3,12 +3,12 @@
 from ray.dag import DAGNode
 from ray.dag.format_utils import get_dag_node_str
 from ray.experimental.gradio_utils import type_to_string
-from ray.util.annotations import DeveloperAPI
+from ray.util.annotations import Deprecated
 
 IN_CONTEXT_MANAGER = "__in_context_manager__"
 
 
-@DeveloperAPI
+@Deprecated(message="The DAG API is deprecated.")
```

Comment:
same for these, should point to replacement

for the most part when we deprecate, best to point people in the right direction